### ISSUE TITLE
Lobby match results

### ISSUE THREAD

**Issue by danielvieira43** [Arcana/node-dota2#583]
I'm using arcana bot successfully keeping it in the lobby until the game ends to get match result. With the last  "practiceLobbyUpdate" I can register all ranking data for players but I think keep the bot waiting the game end is waste of expensive resource. What's your suggestion to get the match result after game ends without keeping the bot in the lobby until the game ends?

*Another Help me tag

Thanks a lot!

**Comment by angel200sdnot** [comment]
If you use a league ticket for the lobby then the result will be available online on the Dota 2 API

**Comment by danielvieira43** [comment]
Yes, but I'm looking for a event or trigger to auto update ranking points right after the game ends. I know there is no magic, but maybe a cool tip or trick to do that. Spamming dota 2 API until ir returns the game result? Feels bad also. Thanks Angel!

**Comment by Crazy-Duck** [comment]
There is unfortunately no other way. If you keep your bot in the lobby until the end of the game, you can detect the end of the game and then query the api (again, only if the lobby is ticketed!). Other than that, the only way to know is to keep spamming the api until the result becomes available. Valve has no subscription api or anything where you can ask to get a notification when match results become available

**Comment by angel200sdnot** [comment]
There is a way of doing this, when the match finishes the bot receives a message from the game coordinator with the match id of the match that finished, this means that the bot can create a lobby, launch it and abandon the match so he is free to create a new lobby. You can have 5 matches going on at the same time that were created by the same bot and the bot will still receive this message when those matches finish. This event is not part of this library so you will have to add it yourself, you can call it "matchSignedOut"

I used to do this because i also didn't want to make an api call every minute to check for the match result, by using this "matchSignedOut" event i would fetch the api as soon as the match was over and only once

**Comment by danielvieira43** [comment]
I get the 'UNHANDLED MESSAGE k_EMsgGCToClientMatchSignedOut' and I will do some test to see what happens. Thanks a lot for your answer! Cya!

**Comment by lemkova** [comment]
hi, just wondering how did u get the match id from the lobby? 
thanks.

**Comment by Crazy-Duck** [comment]
@danielvieira43 @angel200sdnot feel free to create a pull request adding a handler for this event!

**Comment by st0n3vn** [comment]
@lemkova Once the match has started, it is part of the lobby snapshot received on practiceLobbyUpdate as match_id.

**Comment by VDK64** [comment]
I have added my new ticket to the lobby and started server, but don't see statistics in dota api after match. Please, help me how it work. Maybe i do some mistakes?
